### ISSUE TITLE
solidity binaries always redownloaded on nixos

### ISSUE THREAD

**Issue by d-xo** [alloy-rs/svm-rs#94]
On nixos solidity binaries are always downloaded on every invocation of a `forge` command. I'm guessing this is due to some hash check failure as a result of the patching performed since: https://github.com/alloy-rs/svm-rs/pull/48.

I'm wondering what the best way to fix this is?

**Comment by mattsse** [comment]
does this change the binary:

https://github.com/alloy-rs/svm-rs/pull/48/files#diff-b1a35a68f14e696205874893c07fd24fdb88882b47c23cc0e0c80a30c7d53759R82-R84

? if so then this would explain a hash mismatch

**Comment by d-xo** [comment]
yes that modifies the binary. We could perhaps precompute the expected hashes on nixos and ship them as part of svm-rs?

**Comment by mattsse** [comment]
I like that.

one concern I have re additional solc bins, is that this makes adding new binaries more complex, but https://github.com/hellwolf/solc.nix looks well maintained and we could just use this for nix, so pulling from https://github.com/hellwolf/solc.nix would be easier to impl and the update overhead is still manageable.

but I'm fine with either solution.

**Comment by d-xo** [comment]
Using an external nix repo is probably the easiest to implement, but I'm a bit worried about supply chain risk and introducing additional trusted parties.
